pkg/operator/ceph/object/health.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package object
+
+import (
+	"crypto/md5"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	s3UserHealthCheckName      = "rook-ceph-internal-s3-user-checker"
+	s3HealthCheckBucketName    = "rook-ceph-internal-s3-bucket-checker"
+	defaultStatusCheckInterval = 1 * time.Minute
```

Comment:
It's a fairly heavy operation to create a user, bucket, and perform a PUT and GET. The test user and bucket will also remain permanently, right? If the admin doesn't want that user and bucket to ever exist, they should be able to disable the health check. So perhaps we need settings in the CR such as:
```
  healthCheck:
    enabled: true
    interval: 60s
```

Reply:
The operation is not heavy because the user/bucket creation only happens ones, then we just PUT/GET an object. We can make this configurable though.

Reply:
I don't mean that it is a heavy operation for performance, but I mean there are side effects. There is a bucket and a user that is created just for the purpose of testing that might not be desired.

Reply:
Given that we are getting more requests to make liveness probe configurable, perhaps that new spec can be generic enough to help with that.